sdk/storage/azure-storage-blob/azure/storage/blob/_shared/uploads_async.py

```diff
@@ -19,6 +19,26 @@
 from .uploads import SubStream, IterStreamer  # pylint: disable=unused-import
 
 
+async def _async_parallel_uploads(uploader, pending, running):
+    range_ids = []
+    while True:
+        # Wait for some download to finish before adding a new one
+        done, running = await asyncio.wait(running, return_when=asyncio.FIRST_COMPLETED)
+        range_ids.extend([chunk.result() for chunk in done])
+        try:
+            for _ in range(0, len(done)):
+                next_chunk = await pending.__anext__()
+                running.add(asyncio.ensure_future(uploader(next_chunk)))
+        except StopAsyncIteration:
+            break
+
+    # Wait for the remaining uploads to finish
+    if running:
+        done, _running = await asyncio.wait(running)
+        range_ids.extend([chunk.result() for chunk in done])
```

Comment:
I'm assuming any errors raised from this get handled further up the stack?

Reply:
I think so, yeah. This is pretty much a copy from the existing helper, just made to support an async generator which has existed for some time, so I have to assume error handling is correct :). Either way, this is all wrapped in try/except in the client that should handle any errors appropriately.